app/models/refer/referral_code.rb

```diff
@@ -7,7 +7,7 @@ class ReferralCode < ApplicationRecord
     validates :code, presence: true, uniqueness: true
 
     before_validation if: -> { Refer.code_generator } do
-      self.code = Refer.code_generator.call(referrer)
+      self.code = Refer.code_generator.call(referrer) unless self.code
```

Comment:
```suggestion
      self.code ||= Refer.code_generator.call(referrer)
```

I was thinking this might be cleaner, but I guess wouldn't handle if code was an empty string.